connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -208,6 +223,19 @@ protected void initLoaders() {
         // Finally add parent/system loader.
         initPluginLoader(CLASSPATH_NAME);
         addAllAliases();
+        reportPluginConflicts();
+    }
+
+    //visible for testing
+    Set<String> reportPluginConflicts() {
+        return allAddedPlugins.entrySet().stream().filter(e -> e.getValue().size() > 1).map(e -> {
+            String pluginClassName = e.getKey();
+            PluginDesc<?> usedPluginDesc = usedPluginDesc(pluginClassName);
+            List<PluginDesc<?>> ignoredPlugins = new ArrayList<>(e.getValue());
+            ignoredPlugins.remove(usedPluginDesc);
+            log.error("Detected multiple plugins contain '{}'; using {} and ignoring {}", pluginClassName, usedPluginDesc, ignoredPlugins);
```

Comment:
Since this is an error, it probably would be good to clarify the error message and provide an action. WDYT about:
```suggestion
            log.error("Detected multiple plugins contain '{}'; using plugin {} and ignoring {} plugins ({}). "
                    + "Check the installation and remove duplicate plugins from all workers.",
                    pluginClassName, usedPluginDesc, ignoredPlugins.size(), ignoredPlugins);
```

Reply:
done